src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -242,16 +242,22 @@ bool layout_optimizer::can_fuse_reorder(program_node& prev, program_node& next,
     if (next.is_type<reorder>())
         return true;
 
+    auto found_item = std::find_if(next.get_dependencies().begin(), next.get_dependencies().end(),
+                                    [=](cldnn::program_node* node) { return node->is_type<reorder>(); });
+    if (found_item == std::end(next.get_dependencies())) {
+        return false;
+    }
+    auto& reorder_node = found_item[0]->as<reorder>();
+
     // Errata for onednn layout selection
-    if (next.is_type<convolution>() && next.get_dependencies().size() == 1 &&
+    if (next.is_type<convolution>() && next.get_dependency_index(reorder_node) == 0 &&
```

Comment:
need unit test for that

Reply:
Could you explain which unittest you mean? Do you expect to have unittest for every `if` condition of `can_fuse_reorder` function? I'm afraid that might not pay off because it is difficult to make and to maintain..

Reply:
> Do you expect to have unittest for every if condition of can_fuse_reorder function?

Ideally - yes. Realistically, this particular fix is needed for some model probably, so I think it's not that hard to add single test with the pattern that reproduces the issue.